src/runtime/Rule.java

```diff
@@ -407,7 +407,6 @@ public long allLockFailures()
 	public long lockfailure = 0;
 
 	Benchmark(Thread thread) {
-		this.threadid = (Env.majorVersion == 1 && Env.minorVersion > 4) 
-		? thread.getId() : thread.hashCode();
+		thread.getId();
```

Comment:
```suggestion
		this.threadid = thread.getId();
```
元のコードに従うならこちらの方が良いと思います